docs/reference/snapshot-restore/apis/put-repo-api.asciidoc

```diff
@@ -0,0 +1,260 @@
+a response. If no response is received before the timeout expires, the request
+fails and returns an error. Defaults to `30s`.
++
+IMPORTANT: You can also specify this value using the `timeout` query
+parameter. If both parameters are specified, only the query parameter is used.
+
+[[put-snapshot-repo-api-request-type]]
+`type`::
++
+--
+(Required, string)
+Repository type.
+
+.Valid values for `type`
+[%collapsible%open]
+====
+`fs`::
+Shared file system repository. Repositories of this type use a shared file
+system to store snapshots. This file system must accessible to all master and
+data nodes in the cluster.
++
+IMPORTANT: To register a shared file system repository, you must mount the same
+shared filesystem to the same location on all master and data nodes. This
+location must be registered in the `path.repo` setting on all master and data
+nodes in the cluster.
++
+See <<snapshots-filesystem-repository>>.
+
+[xpack]#`source`#::
+Source-only repository. You can use source-only repositories to create minimal,
+source-only snapshots that take up to 50% less space on disk.
++
+Source-only snapshots are only supported if the <<mapping-source-field,`_source`
+field>> is enabled and no
+<<request-body-search-source-filtering,source-filtering>> is applied.
++
+WARNING: Source-only snapshots contain stored fields and index metadata. They do
+not include index or doc values structures and are not searchable when restored.
+After restoring a source-only snapshot, you must <<docs-reindex,reindex>> the
+data into a new index.
++
+See <<snapshots-source-only-repository>>.
+
+`url`::
+URL repository. Repositories of this type are read-only
+for the cluster. This means the cluster can retrieve or restore snapshots from
+the repository but cannot write or create snapshots in it.
++
+You can use URL repositories as an alternative way to give a cluster read-only
+access to a shared file system (`fs`) repository.
++
+See <<snapshots-read-only-repository>>.
+====
+
+More repository types are available through these official
+plugins:
+
+* {plugins}/repository-s3.html[repository-s3] for S3 repository support
+* {plugins}/repository-hdfs.html[repository-hdfs] for HDFS repository support in
+  Hadoop environments
+* {plugins}/repository-azure.html[repository-azure] for Azure storage
+  repositories
+* {plugins}/repository-gcs.html[repository-gcs] for Google Cloud Storage
+  repositories
+--
+
+`settings`::
++
+--
+(Required, object)
+Contains settings for the repository. Valid properties for the `settings` object
+depend on the repository type, set using the
+<<put-snapshot-repo-api-request-type,`type`>> parameter.
+
+.Valid `settings` properties for `fs` repositories
+[%collapsible%open]
+====
+`chunk_size`::
+(Optional, <<byte-units,byte value>>)
+Maximum size of files in snapshots. In snapshots, files larger than this are
+broken down into chunks of this size or smaller. Defaults to `null` (unlimited
+file size).
+
+`compress`::
```

Comment:
Should we just leave this out of the documentation maybe? This is a really useless setting at this point. Having it documented has led to some confusion in the past and I can't see what a user would get out of setting this to `false`.
@DaveCTurner @tlrx any thoughts on this?

Reply:
I'll wait for more feedback, but if we remove this, I'll also remove the reference here:
https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshots-register-repository.html#snapshots-filesystem-repository